source/include/stun_serializer.h

```diff
@@ -0,0 +1,118 @@
+                                                        uint64_t tieBreaker );
+
+StunResult_t StunSerializer_AddAttributeUsername( StunContext_t * pCtx,
+                                                  const char * pUsername,
+                                                  uint16_t usernameLength );
+
+StunResult_t StunSerializer_AddAttributeData( StunContext_t * pCtx,
+                                              const char * pData,
+                                              uint16_t dataLength );
+
+StunResult_t StunSerializer_AddAttributeRealm( StunContext_t * pCtx,
+                                               const char * pRealm,
+                                               uint16_t realmLength );
+
+StunResult_t StunSerializer_AddAttributeNonce( StunContext_t * pCtx,
+                                               const char * pNonce,
+                                               uint16_t nonceLength );
+
+StunResult_t StunSerializer_AddAttributeRequestedTransport( StunContext_t * pCtx,
+                                                            const char * pRequestedTransport,
+                                                            uint16_t requestedTransportLength );
+
+StunResult_t StunSerializer_AddAttributeIntegrity( StunContext_t * pCtx,
+                                                   const char * pIntegrity,
+                                                   uint16_t integrityLength );
+
+StunResult_t StunSerializer_AddAttributeAddress( StunContext_t * pCtx,
+                                                 StunAttributeAddress_t * pstunMappedAddress,
+                                                 uint8_t * pTransactionId,
+                                                 StunAttributeType_t attributeType );
+
+StunResult_t StunSerializer_AddAttributeMappedAddress( StunContext_t * pCtx,
+                                                       StunAttributeAddress_t * pstunMappedAddress,
+                                                       uint8_t * pTransactionId );
+
+StunResult_t StunSerializer_AddAttributeResponseAddress( StunContext_t * pCtx,
+                                                         StunAttributeAddress_t * pstunMappedAddress,
+                                                         uint8_t * pTransactionId );
+
+StunResult_t StunSerializer_AddAttributeSourceAddress( StunContext_t * pCtx,
+                                                       StunAttributeAddress_t * pstunMappedAddress,
+                                                       uint8_t * pTransactionId );
+
+StunResult_t StunSerializer_AddAttributeChangedAddress( StunContext_t * pCtx,
+                                                        StunAttributeAddress_t * pstunMappedAddress,
+                                                        uint8_t * pTransactionId );
+
+StunResult_t StunSerializer_AddAttributeChangedReflectedFrom( StunContext_t * pCtx,
```

Comment:
NIT : For Deserializer , this API is `StunDeserializer_ParseAttributeReflectedFrom` , we can keep the same name for consistency ?

Reply:
The APIs are doing 2 different things, one is parsing the incoming attribute, where as other is adding the attribute, hence the names. I think its fine, let me know if you think otherwise.

Reply:
Sure will remove the Changed keyword.